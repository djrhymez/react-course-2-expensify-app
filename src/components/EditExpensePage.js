import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  }

  onDelete = () => {
    this.props.startRemoveExpense({id: this.props.expense.id});
    this.props.history.push('/');
  }

  render() {
    return (
      <div>
        <ExpenseForm 
          expense={this.props.expense}
          onSubmit={this.onSubmit}
        />
        <button 
          onClick={this.onDelete}
        >Remove</button>
      </div>
    );
  }
};

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    startEditExpense: (expenseId, expense) => dispatch(startEditExpense(expenseId, expense)),
    startRemoveExpense: (expense) => dispatch(startRemoveExpense(expense))
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);