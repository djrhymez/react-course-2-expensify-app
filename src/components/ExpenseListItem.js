import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({id, description, amount, createdAt, dispatch}) => (
  <div>
    <Link to={`/edit/${id}`}><h2>{description}</h2></Link>  
    <p>Amount: {amount}</p>
    <p>Created At: {createdAt}</p>
  </div>
);

export default ExpenseListItem;