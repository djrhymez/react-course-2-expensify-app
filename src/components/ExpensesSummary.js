import React from 'react';
import {connect} from 'react-redux';
import getVisibleExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = ({expenseCount, expensesTotal}) => {
  const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
  const formattedTotal = numeral(expensesTotal / 100).format('$0,0.00');
  return (
    <div>
      <h1>
        Viewing {expenseCount} {expenseWord} totalling {formattedTotal}
      </h1>
    </div>
  );
};

const mapPropsToState = (state) => {
  const visibleExpenses = getVisibleExpenses(state.expenses, state.filters);

  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: getExpensesTotal(visibleExpenses) 
  };
};

export default connect(mapPropsToState)(ExpensesSummary);