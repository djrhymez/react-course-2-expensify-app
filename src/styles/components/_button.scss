.button {
  border: none;
  font-size: $font-size-large;
  padding: $s-size;
  background-color: $blue;
  color: white;
  border-radius: 2px;
  font-weight: 100;
  border-bottom: 2px solid darken($color: $blue, $amount: 10%);
  display: inline-block;
  text-decoration: none;
  line-height: 1;
}

.button--link {
  background: none;
  border: none;
}

.button--secondary {
  background: #888888;
  border: none;
  border-bottom: 2px solid darken($color: #888888, $amount: 10%);
}