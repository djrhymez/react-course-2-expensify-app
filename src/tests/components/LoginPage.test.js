import {LoginPage} from '../../components/LoginPage';
import React from 'react';
import { shallow } from 'enzyme';

let wrapper, startLogin;

beforeEach(() => {
  startLogin = jest.fn();
  wrapper = shallow(<LoginPage startLogin={startLogin}/>);  
});

test('should render login page properly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should call startLogin on button click', () => {
  wrapper.find('button').at(0).simulate('click');
  expect(startLogin).toHaveBeenCalledTimes(1);
});