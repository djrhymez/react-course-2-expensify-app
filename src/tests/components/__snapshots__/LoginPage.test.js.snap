// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render login page properly 1`] = `
<div>
  <button>
    Login
  </button>
</div>
`;
